src/frequenz/sdk/microgrid/component_data.py

```diff
@@ -17,229 +19,209 @@
+        +ve current means consumption, away from the grid.
+        -ve current means supply into the grid.
 
-    @property
-    def voltage_per_phase(self) -> Tuple[float, float, float]:
-        """Get voltages measured by the meter per phase.
+    voltage_per_phase: the AC voltage in Volts (V) between the line and the neutral
+        wire for phase/line 1,2 and 3 respectively.
+    """
 
-        Returns:
-            Voltage values for each phase.
-        """
-        phase_1 = self._raw.meter.data.ac.phase_1.voltage.value
-        phase_2 = self._raw.meter.data.ac.phase_2.voltage.value
-        phase_3 = self._raw.meter.data.ac.phase_3.voltage.value
-        return (phase_1, phase_2, phase_3)
-
-
-class BatteryData(_BaseComponentData):
-    """A wrapper class for holding battery data."""
+    active_power: float
+    current_per_phase: Tuple[float, float, float]
+    voltage_per_phase: Tuple[float, float, float]
 
-    def __init__(self, raw: microgrid_pb.ComponentData) -> None:
-        """Create a `BatteryData` instance.
+    @staticmethod
+    def from_proto(raw: microgrid_pb.ComponentData) -> MeterData:
+        """Create MeterData from a protobuf message.
 
         Args:
-            raw: raw component data as decoded from the wire.
-        """
-        self._raw = raw
-
-    @property
-    def soc(self) -> float:
-        """Get current SoC level.
+            raw: protobuf message
 
         Returns:
-            SoC level.
+            Instance of MeterData created from the protobuf message.
         """
-        return self._raw.battery.data.soc.avg
+        return MeterData(
+            component_id=raw.id,
+            timestamp=raw.ts.ToDatetime(tzinfo=pytz.UTC),
+            active_power=raw.meter.data.ac.power_active.value,
+            current_per_phase=(
+                raw.meter.data.ac.phase_1.current.value,
+                raw.meter.data.ac.phase_2.current.value,
+                raw.meter.data.ac.phase_3.current.value,
+            ),
+            voltage_per_phase=(
+                raw.meter.data.ac.phase_1.voltage.value,
+                raw.meter.data.ac.phase_2.voltage.value,
+                raw.meter.data.ac.phase_3.voltage.value,
+            ),
+        )
 
-    @property
-    def soc_upper_bound(self) -> float:
-        """Get SoC upper bound.
 
-        Returns:
-            Maximum SoC level.
-        """
-        return self._raw.battery.data.soc.system_bounds.upper
+@dataclass(frozen=True)
+class BatteryData(ComponentData):
+    """A wrapper class for holding battery data.
 
-    @property
-    def soc_lower_bound(self) -> float:
-        """Get SoC lower bound.
+    soc: battery's overall SoC in percent (%).
 
-        Returns:
-            Minimum SoC level.
-        """
-        return self._raw.battery.data.soc.system_bounds.lower
+    soc_lower_bound: the SoC below which discharge commands will be blocked by the
+        system.
```

Comment:
I would still add this is in percentage also explicitly to avoid any chance of confusion (and the same for all bounds).

Reply:
Done